src/cursor/find_cursor.ts

```diff
@@ -412,8 +412,15 @@ export class FindCursor<TSchema = any> extends AbstractCursor<TSchema> {
    * @remarks
    * {@link https://docs.mongodb.com/manual/reference/command/find/#find-cmd-allowdiskuse | find command allowDiskUse documentation}
    */
-  allowDiskUse(): this {
+  allowDiskUse(allow = true): this {
     assertUninitialized(this);
+
+    // As of 6.0 the default is true. This allows users to get back to the old behaviour.
```

Comment:
We should move this check to below the check for `sort`.  

Can we also add tests confirming that a MongoInvalidArgumentError is thrown if the user attempts to set allowDiskUse without a sort on their query?